SingularityService/src/main/java/com/hubspot/singularity/resources/HistoryResource.java

```diff
@@ -107,9 +110,42 @@ public SingularityDeployHistory getDeploy(@ApiParam("Request ID for deploy") @Pa
   }
 
   @GET
+  @Path("/request/{requestId}/deploy/{deployId}/tasks/active")
+  @ApiOperation("Retrieve the task history for a specific deploy.")
+  public List<SingularityTaskIdHistory> getActiveDeployTasks(
+      @ApiParam("Request ID for deploy") @PathParam("requestId") String requestId,
+      @ApiParam("Deploy ID") @PathParam("deployId") String deployId) {
+
+    List<SingularityTaskId> taskIds = new ArrayList<SingularityTaskId>();
+    for (SingularityTaskId id : taskManager.getActiveTaskIdsForRequest(requestId)) {
+      if (id.getDeployId().equals(deployId)) {
+        taskIds.add(id);
+      }
+    }
+    return taskHistoryHelper.getHistoriesFor(taskIds);
+  }
+
+  @GET
+  @Path("/request/{requestId}/deploy/{deployId}/tasks/inactive")
+  @ApiOperation("Retrieve the task history for a specific deploy.")
+  public List<SingularityTaskIdHistory> getInactiveDeployTasks(
+      @ApiParam("Request ID for deploy") @PathParam("requestId") String requestId,
+      @ApiParam("Deploy ID") @PathParam("deployId") String deployId) {
+
+    List<SingularityTaskId> taskIds = new ArrayList<SingularityTaskId>();
+    for (SingularityTaskId id : taskManager.getInactiveTaskIdsForRequest(requestId)) {  
```

Comment:
This isn't good enough, unfortunately. Singularity users can optionally persist non-operational data (i.e. history) into mysql, but this method only hits ZK for data. Consider adding a new `getBlendedHistory()` method in TaskHistoryHelper that accepts both a request and deploy ID.